packages/material-renderers/src/layouts/ExpandPanelRenderer.tsx

```diff
@@ -296,12 +299,35 @@ export const withContextToExpandPanelProps = (
     props,
   }: JsonFormsStateContext & ExpandPanelProps) {
     const dispatchProps = ctxDispatchToExpandPanelProps(ctx.dispatch);
-    const { childLabelProp, schema, path, index, uischemas } = props;
+    const { childLabelProp, schema, rootSchema, path, index, uischemas } =
+      props;
     const childPath = composePaths(path, `${index}`);
     const childData = Resolve.data(ctx.core.data, childPath);
-    const childLabel = childLabelProp
-      ? get(childData, childLabelProp, '')
-      : get(childData, getFirstPrimitiveProp(schema), '');
+
+    let childLabel;
+    if (childLabelProp) {
+      childLabel = get(childData, childLabelProp, '');
+
+      const childSchema = Resolve.schema(
+        schema,
+        `#/properties/${encode(childLabelProp)}`,
+        rootSchema
+      );
+      if (
+        childSchema &&
+        childSchema.type === 'string' &&
+        (childSchema.enum !== undefined || childSchema.const !== undefined)
```

Comment:
Currently we only check for the `enum` case but we should also cover the "one of enums" which are pairs of const an optional title properties, similar to the testers of the enum and oneofenum controls.

Then we need to call the `enumToEnumOptionMapper` or the `oneOfToEnumOptionMapper` accordingly.

Reply:
I don't have case of enumOf on my side, I'll look to find some in the examples, and add it.

Reply:
Added the enumOf case.

Note: the default behavior when no translating is setup will change (showing `const` value => showing `title` value instead). I think it's better but your mileage may vary.

Reply:
Its definitely better ;)